data/src/test/java/io/gnosis/data/repositories/TransactionRepositoryTest.kt

```diff
@@ -46,322 +117,106 @@ class TransactionRepositoryTest {
+                        assertEquals((pagedResult[i].txInfo as Custom).dataSize, "${dataSize}")
+                    }
+                    is SettingsChange -> {
+                        assertEquals(pagedResult[i].executionInfo?.nonce, (this as Transaction.SettingsChange).nonce)
+                        assertEquals(pagedResult[i].txStatus, this.status)
+                        assertEquals((pagedResult[i].txInfo as SettingsChange).dataDecoded, dataDecoded)
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    fun `loadTransactionsPage (all tx type) should return respective tx type`() = runBlockingTest {
+        val pagedResult = listOf(
+            buildGateTransactionDto(txInfo = buildTransferTxInfo()),
+            buildGateTransactionDto(txInfo = buildCustomTxInfo()),
+            buildGateTransactionDto(txInfo = buildSettingsChangeTxInfo())
+        )
+        coEvery { gatewayApi.loadTransactionsPage(any()) } returns Page(1, null, null, pagedResult)
+
+        val actual = transactionRepository.loadTransactionsPage("url")
+
+        assertEquals(3, actual.results.size)
+
+        (0..2).forEach { i ->
+            with(actual.results[i]) {
+                when (pagedResult[i].txInfo) {
+                    is Transfer -> {
+                        assertEquals(pagedResult[i].executionInfo?.nonce, (this as Transaction.Transfer).nonce)
+                        assertEquals(pagedResult[i].txStatus, this.status)
+                        assertEquals((pagedResult[i].txInfo as Transfer).direction == TransactionDirection.INCOMING, incoming)
+                    }
+                    is Custom -> {
+                        assertEquals(pagedResult[i].executionInfo?.nonce, (this as Transaction.Custom).nonce)
+                        assertEquals(pagedResult[i].txStatus, this.status)
+                        assertEquals((pagedResult[i].txInfo as Custom).dataSize, "${dataSize}")
+                    }
+                    is SettingsChange -> {
+                        assertEquals(pagedResult[i].executionInfo?.nonce, (this as Transaction.SettingsChange).nonce)
+                        assertEquals(pagedResult[i].txStatus, this.status)
+                        assertEquals((pagedResult[i].txInfo as SettingsChange).dataDecoded, dataDecoded)
+                    }
+                }
+            }
+        }
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun `getTransactions (unknown txInfo) should throw exception Transfers`() = runBlockingTest {
+        val pagedResult = listOf(
+            buildGateTransactionDto(txInfo = object : TransactionInfo {
+                override val type: GateTransactionType
+                    get() = GateTransactionType.Transfer
+            })
+        )
+        coEvery { gatewayApi.loadTransactions(any()) } returns Page(1, null, null, pagedResult)
+
+        transactionRepository.getTransactions(defaultSafeAddress)
```

Comment:
If we use `@Test(expected="...")` we can't really verify what's internally call within the mock, as the method exists early on the throw. I am much more inclined to to use `runCatching` then have additionally a `coVerify` block where we see that we don't call a method more times than we should, etc. And we still can look into the exception.